src/Microsoft.ComponentDetection.Contracts/TypedComponent/ConanComponent.cs

```diff
@@ -9,16 +9,27 @@ private ConanComponent()
         // reserved for deserialization
     }
 
-    public ConanComponent(string name, string version)
+    public ConanComponent(string name, string version, string previous, string packageId)
     {
         this.Name = this.ValidateRequiredInput(name, nameof(this.Name), nameof(ComponentType.Conan));
         this.Version = this.ValidateRequiredInput(version, nameof(this.Version), nameof(ComponentType.Conan));
+        this.Md5Hash = this.ValidateRequiredInput(previous, nameof(this.Md5Hash), nameof(ComponentType.Conan));
+        this.Sha1Hash = this.ValidateRequiredInput(packageId, nameof(this.Sha1Hash), nameof(ComponentType.Conan));
+        this.Supplier = "ConanCenter";
     }
 
     public string Name { get; set; }
 
     public string Version { get; set; }
 
+    public string Md5Hash { get; set; }
+
+    public string Sha1Hash { get; set; }
+
+    public string Supplier { get; }
+
+    public string PackageSourceURL => $"https://conan.io/center/recipes/{this.Name}?version={this.Version}";
```

Comment:
Are these two necessary? I definitely agree with the hashes.

The supplier is always `ConanCenter` and the `PackageSourceUrl` can be computed by the end user. I don't believe we store the source URL in any typed component.

Reply:
1) PackageSourceURL is actually the download URL which end user can compute definitely.. But since, we have corresponding attribute "downloadLocation", I believe it is better to calculate and assign this so that end user can navigate to the url with one click. 

example of data

`{
      "name": "jwt-cpp",
      "SPDXID": "SPDXRef-Package-38296822FEC0FAB8266C8F2C99ABBDA7BB985BA19F82BDF9CD0C1A86A8D76751",
      "downloadLocation": "https://conan.io/center/recipes/jwt-cpp?version=0.6.0#9ba33a57e1b44045f50633246c660de4",
      "filesAnalyzed": false,
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "NOASSERTION",
      "copyrightText": "NOASSERTION",
      "versionInfo": "0.6.0#9ba33a57e1b44045f50633246c660de4",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceType": "purl",
          "referenceLocator": "pkg:conan//jwt-cpp@0.6.0#9ba33a57e1b44045f50633246c660de4#"
        }
      ],
      "supplier": "Conancenter"
 }`
 
2) I agrees that supplier is hard coded value as **Conancenter** and can be left as "NOASSERTION" .